service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentMigrationService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class RespondentMigrationService {
+
+    public AboutToStartOrSubmitCallbackResponse setMigratedValue(CaseDetails caseDetails) {
```

Comment:
Why we need that method and associated callback? Is it to make the show / hide condition work properly while `respondentsMigrated` does not exist prior migration?

Reply:
because respondentsMigrated is a hidden field it only exists for the lifecycle of the event. It needs to be set every time you enter the event in order to show the right respondent structure. I extracted to a separate service to unit test that this is working as expected